File: leetcode/solutions/0283.MoveZeroes/moveZeroes_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import moveZeroes from "./moveZeroes.ts";

Deno.test("0283. Move Zeroes", () => {
  type test = [
    number[],
    number[],
  ];
  const tests: test[] = [
		[[0], [0]],
		[[1], [1]],
		[[0, 0, 0], [0, 0, 0]],
		[[0, 1, 0], [1, 0, 0]],
    [[3, 2, 1], [3, 2, 1]],
    [[0, 1, 0, 3, 12], [1, 3, 12, 0, 0]],
  ]

  for (const [arg1, want] of tests) {
    moveZeroes(arg1);
    assertEquals(arg1, want);
  }
});

File: leetcode/solutions/0283.MoveZeroes/moveZeroes_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0283_MoveZeroes(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out []int
	}{
		{[]int{0}, []int{0}},
		{[]int{1}, []int{1}},
		{[]int{0, 0, 0}, []int{0, 0, 0}},
		{[]int{0, 1, 0}, []int{1, 0, 0}},
		{[]int{3, 2, 1}, []int{3, 2, 1}},
		{[]int{0, 1, 0, 3, 12}, []int{1, 3, 12, 0, 0}},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			moveZeroes(tt.in)

			if !cmp.Equal(tt.in, tt.out) {
				t.Errorf("got %#v want %#v", tt.in, tt.out)
			}
		})
	}
}
